sequencer/coordinator.py

```diff
@@ -21,63 +22,82 @@ def __init__(self, zookeeper_connect, autoscaling_grp_name, strategy_name, insta
-                                                 self.max_sequence_id)
+        return result
 
-                    sequence_id = generator.get_sequence_id()
-                    current_instance_node_path = "/".join([instances_root_path, str(sequence_id)])
-                    if not self.interrupted:
-                        try:
-                            self.zk.create(path=current_instance_node_path,
-                                           value=str.encode(str(self.instance_id)),
-                                           ephemeral=False,
-                                           makepath=True)
-                        except NodeExistsError:
-                            self.zk.delete(current_instance_node_path)
-                            self.zk.create(path=current_instance_node_path,
-                                           value=str.encode(str(self.instance_id)),
-                                           ephemeral=False,
-                                           makepath=True)
+    def operation(self):
+        instances_root_path = "/".join([zk_sequencer_root, self.autoscaling_grp_name])
+        self.zk.ensure_path(instances_root_path)
+        instance_nodes = self.zk.get_children(instances_root_path)
+        zk_instance_sequencers = {}
+        for instance_node in instance_nodes:
+            instance_node_path = "/".join([instances_root_path, instance_node])
+            instance_id = self.zk.get(instance_node_path)[0]
+            zk_instance_sequencers[str(instance_id)] = int(instance_node)
 
-                # optimistic concurrency control by assuming that operation would be executed without any
-                # interruption but if any interruption happens, then idempotent operation is executed again
-                if self.interrupted:
-                    logging.debug('distributed sequence coordination was interrupted so retrying with lock')
-                    result = None
-                    continue
-                else:
-                    result = sequence_id
-        finally:
-            self.zk.stop()
+        instance_sequencers = {k: v for k, v in zk_instance_sequencers.items() if
+                               k in self.asg_instances_ids}
+
+        generator = SequenceStrategy(self.strategy_name,
+                                     self.instance_id,
+                                     instance_sequencers,
+                                     self.max_sequence_id)
+
+        sequence_id = generator.get_sequence_id()
+        current_instance_node_path = "/".join([instances_root_path, str(sequence_id)])
+        try:
+            self.zk.create(path=current_instance_node_path,
+                           value=str.encode(str(self.instance_id)),
+                           ephemeral=False,
+                           makepath=True)
+        except NodeExistsError:
+            self.zk.delete(current_instance_node_path)
+            self.zk.create(path=current_instance_node_path,
```

Comment:
Not sure if this is correct way to handle `zk.create` ... What will happen if `zk.create` throws an Error from `NodeExistsError` ?